### PR TITLE
Adventure: Remove unused variables

### PR DESCRIPTION
**Pull request by nicholassaylor**
## What is this fixing or adding?
Removes unused variables and parameters in various locations.

## How was this tested?
Running unittests and generation

### CODE REVIEW

File: worlds/adventure/Regions.py [hunk 1]
```diff
@@ -154,6 +153,6 @@ def create_regions(options: PerGameCommonOptions, multiworld: MultiWorld, player
 
 
 # Placeholder for adding sets of priority locations at generation, possibly as an option in the future
-def determine_priority_locations(world: MultiWorld, dragon_slay_check: bool) -> {}:
+def determine_priority_locations() -> {}:
```
Comment (ScipioWright) [inline worlds/adventure/Regions.py:156]: on the one hand, this is a placeholder so maybe add the parameters to the comment or something
on the other hand, world: MultiWorld is cringe and should be deleted anyway

**Review by ScipioWright — APPROVED** [review]
LGTM, the things removed are clearly unused

File: worlds/adventure/Regions.py (hunk 1 above)
Comment (nicholassaylor) [reply worlds/adventure/Regions.py]: I put the old function def in comments with `world: MultiWorld` changed

**Comment by nicholassaylor** [comment]
Requesting review from @JusticePS

**Review by JusticePS — APPROVED** [review]

**Review by NewSoupVi — APPROVED** [review]

**Pull request merged**